index.js

```diff
@@ -48,50 +48,274 @@ const player = {
     { id: 5, name: 'Israeli', songs: [4, 5] },
   ],
   playSong(song) {
-    console.log(/* your code here */)
+    return `Playing ${song.title} from ${song.album} by ${song.artist} | ${convertSecondsToMinutes(song.duration)}.`
   },
 }
 
 function playSong(id) {
-  // your code here
+  let song = getSongById(id)
+  console.log(player.playSong(song))
 }
 
 function removeSong(id) {
-  // your code here
+  //remove song from songs
+  player.songs.splice(getSongIndexById(id), 1)
+
+  // remove from playlists
+  for (let playlist of player.playlists) {
+    for (let songID of playlist.songs) {
+      if (songID === id) {
+        playlist.songs.splice(playlist.songs.indexOf(id), 1)
+      }
+    }
+  }
 }
 
-function addSong(title, album, artist, duration, id) {
-  // your code here
+function addSong(
+  title,
+  album,
+  artist,
+  duration,
+  id = getVacantId(player.songs)
```

Comment:
Great use of default parameter values!
However, I would recommend that you put this inside the function. I, for one, am unsure if this is evaluated only once when the function is declared, or every time the function is called (and I would bet on the first option). It might mean that your "generated" value is actually always the same...

Reply:
Thanks for the comment! After testing, seems like the function get evaluated every time it is called and returns a unique ID. 
![image](https://user-images.githubusercontent.com/86828178/134587934-dfcb1afd-1a73-4cfd-8355-1813dfcbf83e.png)